### ISSUE TITLE
mkdir() is not working under Python 3.11 (with authentication)

### ISSUE THREAD

**Issue by hagenw** [devopshq/artifactory#404]
I start with an empty repository called `unittests-public` under https://audeering.jfrog.io/artifactory
when I execute the following code it works under Python 3.8, 3.9, 3.10:

```python
import artifactory

path = artifactory.ArtifactoryPath(
    'https://audeering.jfrog.io/artifactory/unittests-public/com/audeering/audfactory',
    auth=(
        'audeering-unittest',
        '<APIKEY>',  # replace with needed API key
    ),
)
path.parent.mkdir()
```

under Python 3.11 I get instead

```
---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
Cell In[5], line 1
----> 1 path.parent.mkdir()

File /usr/lib/python3.11/pathlib.py:1116, in Path.mkdir(self, mode, parents, exist_ok)
   1112 """
   1113 Create a new directory at this given path.
   1114 """
   1115 try:
-> 1116     os.mkdir(self, mode)
   1117 except FileNotFoundError:
   1118     if not parents or self.parent == self:

FileNotFoundError: [Errno 2] No such file or directory: 'https://audeering.jfrog.io/artifactory/unittests-public/com/audeering'
```

**Comment by mdhaisne** [comment]
Hello,
Is there any update regarding this issue ?
I'm having a similar issue with `ArtifactoryPath.glob` which works well on 3.7 to 3.10. But on Python3.11 also raising a `FileNotFoundError`.

**Comment by mdhaisne** [comment]
Oh indeed you are right, I've missed this issue sorry.

**Comment by RichardTea** [comment]
This is still an issue in release 0.9.0.
PR #376 says it added support, but this was incomplete.
PR #424 looks to be the actual fix, hopefully this can be released soon?

**Comment by allburov** [comment]
@RichardTea thank you for noticing it, published new version with the fix! https://github.com/devopshq/artifactory/releases/tag/0.9.1

**Comment by hagenw** [comment]
Thank you very much for implementing a fix.
I can confirm that the issue is solved with release 0.9.1.
